missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/geometry/GameArea.java

```diff
@@ -0,0 +1,121 @@
+
+    private int minX, minY, minZ;
+    private int maxX, maxY, maxZ;
+
+    public GameArea(Location pos1, Location pos2) {
+        
+        if (!Geometry.bothLocInSameWorld(pos1, pos2)) throw new IllegalArgumentException("Defined positions are not in the same world!");
+        if (pos1 == pos2) throw new IllegalArgumentException("The selected positions do not differ.");
+        
+        this.world = pos1.getWorld();
+        
+        this.position1 = pos1;
+        this.position2 = pos2;
+        
+        calculate();
+    }
+    
+    public GameArea(Location center, int offset, int high) {
+        
+        if (offset < 1) throw new IllegalArgumentException("The offset must be higher than 0.");
+        if (high < 2) throw new IllegalArgumentException("The high must be higher than 1.");
+        
+        this.world = center.getWorld();
+        
+        long x1 = center.getBlockX() + offset;
+        long x2 = center.getBlockX() - offset;
+        long z1 = center.getBlockZ() + offset;
+        long z2 = center.getBlockZ() - offset;
+
+        long y1 = center.getBlockY();
+        long y2 = y1 + high;
+
+        this.position1 = new Location(center.getWorld(), x1, y1, z1);
+        this.position2 = new Location(center.getWorld(), x2, y2, z2);
+
+        calculate();
+    }
+    
+    public GameArea(World world, AreaConfiguration areaConfig) {
+        
+        this.world = world;
+        
+        this.position1 = new Location(world, areaConfig.getMinX(), areaConfig.getMinY(), areaConfig.getMinZ());
+        this.position2 = new Location(world, areaConfig.getMaxX(), areaConfig.getMaxY(), areaConfig.getMaxZ());
+
+        if (position1 == position2) throw new IllegalArgumentException("The selected positions do not differ.");
+        
+        calculate();
+    }
+
+    private void calculate() {
+
+        if (position1.getBlockX() < position2.getBlockX()) {
+            minX = position1.getBlockX();
+            maxX = position2.getBlockX();
+        } else {
+            maxX = position1.getBlockX();
+            minX = position2.getBlockX();
+        }
+
+        if (position1.getBlockY() < position2.getBlockY()) {
+            minY = position1.getBlockY();
+            maxY = position2.getBlockY();
+        } else {
+            maxY = position1.getBlockY();
+            minY = position2.getBlockY();
+        }
+
+        if (position1.getBlockZ() < position2.getBlockZ()) {
+            minZ = position1.getBlockZ();
+            maxZ = position2.getBlockZ();
+        } else {
+            maxZ = position1.getBlockZ();
+            minZ = position2.getBlockZ();
+        }
+    }
+
+    public AreaConfiguration getAreaConfiguration() {
+        AreaConfiguration newAreaConfig = new AreaConfiguration(position1.getBlockX(), position1.getBlockY(), position1.getBlockZ(), 
+                position2.getBlockX(), position2.getBlockY(), position2.getBlockZ());
+        return newAreaConfig;
+    }
+
+    private enum Direction {
+
```

Comment:
The `Direction` enum is not currently used, but it fits in well here.

Reply:
Please put this in a comment to the enum, such remarks belong into the code and not just the pull request

Reply:
You can see in the IDE what is not used, right?

Reply:
We should know, why parts of the code where introduced and/or not removed. If I came above an unused enum in a code base, I would look up its purpose/usage/...